src/support/HttpClient.cpp

```diff
@@ -203,3 +208,8 @@ Vault::HttpClient::executeRequest(const Vault::Url& url,
     Vault::HttpResponseBodyString{buffer}
   });
 }
+
+void
+Vault::HttpClient::reportError(std::optional<HttpResponse> response) const {
+    responseErrorCallback_(response.value());
+}
```

Comment:
Since we aren't doing anything other than deferring to the callback, I think it's better to drop this can grab the callback directly now.

Reply:
okay, got rid of the function